### PR TITLE
WS api/metrics/list list metrics - SONAR-6570

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

**Comment by teryk** [comment]
I suggest you review commit by commit as the second one is just a refactoring

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

File: server/sonar-server/src/main/java/org/sonar/server/metric/ws/ListAction.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import org.sonar.server.es.SearchOptions;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.collect.Sets.newHashSet;
+
+public class ListAction implements MetricsWsAction {
```
Comment (jblievremont) [inline server/sonar-server/src/main/java/org/sonar/server/metric/ws/ListAction.java]: Not declared in Level4Components => failure at server startup

Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/metric/ws/ListAction.java]: Seems like the feature branch was not executed

Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/metric/ws/ListAction.java]: yep, I forgot to push this part, sorry

Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/metric/ws/ListAction.java]: @julienlancelot the feature branch has been executed http://jenkins.internal.sonarsource.com/view/wall-feature-branch/job/sq-workflow-server~ws_metrics_list/ there was the error and I would detect it this morning, on my arrival

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Pull request merged**